### PR TITLE
PWX-5477: added events op to create and list events

### PR DESCRIPTION
**Pull request by sdeoras**
This PR adds events op interface and implementation to create and list k8s events

Signed-off-by: Saurabh Deoras <sdeoras@gmail.com>

### CODE REVIEW

File: k8s/k8s.go [hunk 1]
```diff
@@ -70,6 +70,15 @@ type Ops interface {
 	SnapshotOps
 	SecretOps
 	ConfigMapOps
+	EventsOps
```
Comment (piyush-nimbalkar) [inline k8s/k8s.go]: nit: EventOps to be coherent with other interfaces

File: k8s/k8s.go [hunk 2]
```diff
@@ -70,6 +70,15 @@ type Ops interface {
...
+}
+
+// EventsOps is an interface to put and get k8s events
+type EventsOps interface {
+	// CreateEvent puts an event into k8s etcd
+	CreateEvent(event *v1.Event) error
```
Comment (harsh-px) [inline k8s/k8s.go]: Create should return the created event.

File: k8s/k8s.go [hunk 3]
```diff
@@ -2471,6 +2480,17 @@ func (k *k8sOps) UpdateConfigMap(configMap *v1.ConfigMap) (*v1.ConfigMap, error)
 
 // ConfigMap APIs - END
 
+// CreateEvent puts an event into k8s etcd
+func (k *k8sOps) CreateEvent(event *v1.Event) error {
+	_, err := k.client.CoreV1().Events(event.Namespace).Create(event)
```
Comment (piyush-nimbalkar) [inline k8s/k8s.go]: initialize the client before the call.. 
```
if err := k.initK8sClient(); err != nil {
 		return err
  	}

File: k8s/k8s.go (hunk 2 above)
Comment (sdeoras) [reply k8s/k8s.go]: done

File: k8s/k8s.go (hunk 1 above)
Comment (sdeoras) [reply k8s/k8s.go]: done

File: k8s/k8s.go (hunk 3 above)
Comment (sdeoras) [reply k8s/k8s.go]: done

**Pull request merged**